src/ProgressBar/ProgressBar.tsx

```diff
@@ -39,20 +40,24 @@ const ProgressContainer = styled.span<StyledProgressContainerProps>`
 
 export type ProgressBarProps = React.PropsWithChildren & {bg?: string} & StyledProgressContainerProps & ProgressProp
 
-export const ProgressBar = ({
-  progress,
-  bg = 'success.emphasis',
-  barSize = 'default',
-  children,
-  ...rest
-}: ProgressBarProps) => {
-  if (children && progress) {
-    throw new Error('You should pass `progress` or children, not both.')
-  }
-
-  return (
-    <ProgressContainer barSize={barSize} {...rest}>
-      {children ?? <Item progress={progress} sx={{backgroundColor: bg}} />}
-    </ProgressContainer>
-  )
-}
+export const ProgressBar = forwardRef<HTMLSpanElement, ProgressBarProps>(
+  ({progress, bg = 'success.emphasis', barSize = 'default', children, ...rest}: ProgressBarProps, forwardRef) => {
+    if (children && progress) {
+      throw new Error('You should pass `progress` or children, not both.')
+    }
+
+    const ariaAttributes = {
+      'aria-valuenow': typeof progress === 'number' ? Math.round(progress) : undefined,
+      'aria-valuetext': typeof progress === 'string' ? progress : undefined,
+      'aria-valuemin': 0,
+      'aria-valuemax': 100,
+      'aria-busy': progress !== 100,
```

Comment:
Should we consider the string type here as well?

Reply:
@broccolinisoup Thanks for bringing this up. I ended up changing aria attributes a bit since t looks like this component really only works if progress is a number or a string of a number like ("66" or "100") Because the string is really just a number using `aria-valuetext` isn't actually necessary. 

To reduce confusion maybe we should only allow a number passed in?

Reply:
From the first look at the usages at dotcom, in practicality it seems fine but technically narrowing types are considered as `major` change. ([reference](https://github.com/primer/react/blob/main/contributor-docs/versioning.md#:~:text=minor-,The%20type%20of%20a%20prop%20is%20narrowed,-major)) And I don't know if there is any historical reasons to include `string` type in the first place. Sorry @kendallgassner we are dealing with a bit of an old component here I guess and lots to improve - I am trying to keep my review only specific to your changes but seems like some overlaps are inevitable 😬 I'll post internally as well to bring some more reviewers to get more context on this!